### DISCUSSION TITLE
Grouping in METviewer

### DISCUSSION THREAD

**Discussion by hertneky** [dtcenter/METplus#2451] [Plot Generation] [answered]
I would like to look at a number of plots for ensemble output (RHIST, ROC, Reliability, etc), with different lines representing a different grouping of fcst_lead. For example day1 (0-23 lead),.... day7 (144-167). I can make each plot separately just fine. It would be nice to be able to plot all lines on the same plot, but I cannot figure out how to do this. I know of the capability in series plots, to group the Y1 series variables (simple example below)
```
        <series1>
            <field name="model">
                <val>WRF-lag-ensemble</val>
            </field>
            <field name="fcst_init_beg">
                <val>2023-10-16 00:00:00:2023-10-17 00:00:00:2023-10-18 00:00:00</val>
            </field>
        </series1>
```
        
I tried a similar approach in RHIST, for example and received an error when running the script. I was trying to group by lead like below, since I would want to plot a line for day1, day2, etc. What is the correct format for this, or does the capability not exist? (I see no grouping option in the METviewer GUI for these types of plots)
```
        <series1>
            <field name="model">
                <val>WRF-lag-ensemble</val>
	    </field>
            <field name="fcst_lead">
	    	<val>0:10000:20000:30000:40000</val>
	    </field>
            <field name="fcst_lead">
	    	<val>240000:250000:260000:270000:280000</val>
	    </field>
        </series1>
```

**Comment by bikegeek** [comment]
I took a look at a yaml config file for the rank histogram in the METplotpy repo and couldn't figure out how to do this.  I sent an email to @TatianaBurek.  She is out of the office until Friday December 15.

**Comment by bikegeek** [comment]
The ROC diagram does NOT have support for grouping.

**Comment by bikegeek** [comment, marked as answer]
@hertneky, I asked @TatianaBurek  and she confirmed that grouping support is only available in the following plots:

- series
- box
- bar

**Reply by hertneky** [reply]
The project I wanted this capability for has ended, but I could see it being useful for all and should be considered for future development. Thanks.

**Reply by bikegeek** [reply]
I created a Github Issue (enhancement) for future work: https://github.com/dtcenter/METplotpy/issues/423
